static/app/views/issueDetails/traceTimeline/traceTimeline.tsx

```diff
@@ -1,31 +1,46 @@
 import {useRef} from 'react';
 import styled from '@emotion/styled';
 
+import Feature from 'sentry/components/acl/feature';
 import ErrorBoundary from 'sentry/components/errorBoundary';
 import QuestionTooltip from 'sentry/components/questionTooltip';
 import {t} from 'sentry/locale';
 import {space} from 'sentry/styles/space';
 import type {Event} from 'sentry/types/event';
 import useRouteAnalyticsParams from 'sentry/utils/routeAnalytics/useRouteAnalyticsParams';
 import {useDimensions} from 'sentry/utils/useDimensions';
+import useOrganization from 'sentry/utils/useOrganization';
 
 import {TraceTimelineEvents} from './traceTimelineEvents';
-import {useTraceTimelineEvents} from './useTraceTimelineEvents';
+import {EventItem} from './traceTimelineTooltip';
+import {type TimelineEvent, useTraceTimelineEvents} from './useTraceTimelineEvents';
+
+const ISSUES_TO_SKIP_TIMELINE = 2;
 
 interface TraceTimelineProps {
   event: Event;
 }
 
 export function TraceTimeline({event}: TraceTimelineProps) {
+  const organization = useOrganization();
   const timelineRef = useRef<HTMLDivElement>(null);
   const {width} = useDimensions({elementRef: timelineRef});
   const {isError, isLoading, traceEvents} = useTraceTimelineEvents({event});
 
   const hasTraceId = !!event.contexts?.trace?.trace_id;
 
-  let timelineStatus: string | undefined;
+  let timelineStatus: string | undefined = 'empty';
+  let timelineSkipped = false;
+  let issuesCount = 0;
   if (hasTraceId && !isLoading) {
-    timelineStatus = traceEvents.length > 1 ? 'shown' : 'empty';
+    if (!organization.features.includes('related-issues-issue-details-page')) {
+      timelineStatus = traceEvents.length > 1 ? 'shown' : 'empty';
+    } else {
+      issuesCount = getIssuesCountFromEvents(traceEvents);
+      // When we have more than 2 issues regardless of the number of events we skip the timeline
+      timelineSkipped = issuesCount === ISSUES_TO_SKIP_TIMELINE;
```

Comment:
For now, we will only replace the timeline when there are two issues. I have to make some changes here tomorrow.